### PR TITLE
Docx

### PR DESCRIPTION
**Pull request by geramirez**
This PR contains the cli changes for the docx command.
`compliance-masonry-go docs docx -t template.docx -c LATO`

### CODE REVIEW

**Comment by geramirez** [comment]
@jcscottiii this should be ready for review. Sadly, it lowers the coverage by  `0.01%`

File: docs.go [hunk 1]
```diff
@@ -7,43 +7,67 @@ import (
...
 			fileName := strings.TrimSuffix(file.Name(), filepath.Ext(file.Name()))
-			messages = append(messages, fmt.Sprintf("`compliance-masonry-go docs gitbook %s`", fileName))
+			messages = append(messages, fmt.Sprintf("`%s`", fileName))
 		}
+		return "", messages
+	}
+	return certificationPath, messages
+}
+
+func BuildTemplate(config *docx.Config) []string {
```
Comment (jcscottiii) [inline docs.go]: i don't think you need to use a pointer to docx.Config. (None of it's functions are mutators)

File: docs.go [hunk 2]
```diff
@@ -7,43 +7,67 @@ import (
...
+	err := config.BuildDocx()
+	if err != nil {
+		messages = append(messages, err.Error())
+	} else {
+		messages = append(messages, "New Docx Created")
+	}
+	return messages
+}
+
+func MakeGitbook(config *gitbook.Config) []string {
```
Comment (jcscottiii) [inline docs.go]: same

File: docs_test.go [hunk 3]
```diff
@@ -1,89 +1,112 @@
...
-	},
-	{
-		gitbookConfig{
-			opencontrolDir: "fixtures/opencontrol_fixtures_with_markdown/",
-			certification:  "",
-			markdownPath:   "fixtures/opencontrol_fixtures_with_markdown/markdowns/",
-		},
-		[]string{"Error: New Missing Certification Argument", "Usage: masonry-go docs gitbook FedRAMP-low"},
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
```
Comment (jcscottiii) [inline docs_test.go]: you can dot import this if you want. (i did it for most of mine except a package that also had the name `Entry` so I couldn't dot assign it for that test suite. that's up to you though.

File: docx/docx.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+						text += fmt.Sprintf("- %s %s  \n", verification.Name, verification.Path)
+					}
+				})
+			}
+		}
+	})
+	return text
+}
+
+// splitControl returns a split standard and control given a standard
```
Comment (jcscottiii) [inline docx/docx.go]: Capital `S`

File: docx/docx.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		return err
+	}
+	funcMap := template.FuncMap{"getControl": openControl.FormatControl}
+	docTemplate.AddFunctions(funcMap)
+	docTemplate.Parse()
+	docTemplate.Execute(config.ExportPath, nil)
+	return err
+}
+
+// getControl returns a control formatted for docx
```
Comment (jcscottiii) [inline docx/docx.go]: Comment `getControl` to `FormatControl`

File: gitbook/gitbook.go [hunk 6]
```diff
@@ -43,16 +51,16 @@ func replaceParentheses(text string) string {
 }
 
 // BuildGitbook entry point for creating gitbook
-func BuildGitbook(opencontrolDir string, certificationPath string, markdownPath string, exportPath string) {
+func (config *Config) BuildGitbook() {
```
Comment (jcscottiii) [inline gitbook/gitbook.go]: same. you can probably make this just `(config Config)`

**Comment by jcscottiii** [comment]
also check the test step in circle ci. it fails but it still reports a success. 
https://circleci.com/gh/opencontrol/compliance-masonry-go/120

**Comment by geramirez** [comment]
@jcscottiii Ok, I made the updates and changed the tests so they work with the new version of `codegangsta/cli`

**Pull request merged**